core/src/main/java/com/opensymphony/xwork2/conversion/impl/StringConverter.java

```diff
@@ -65,8 +65,17 @@ protected String convertToString(Locale locale, Object value) {
         if (Number.class.isInstance(value)) {
             NumberFormat format = NumberFormat.getNumberInstance(locale);
             format.setGroupingUsed(false);
-            if (Double.class.isInstance(value) || BigDecimal.class.isInstance(value)) {
+            if (BigDecimal.class.isInstance(value) || Double.class.isInstance(value) || Float.class.isInstance(value)) {
                 format.setMinimumFractionDigits(1);
+                if (BigDecimal.class.isInstance(value)) {
+                    format.setMaximumFractionDigits(Integer.MAX_VALUE);
+                }
+                else if (Double.class.isInstance(value)) {
+                    format.setMaximumFractionDigits(15);
```

Comment:
Where `15` comes from? Shouldn't it be `16`?

Reply:
Ouch! I mislead max fraction digits with max precision while they're different. thank you. fixed.